datadog-setup.php

```diff
@@ -47,23 +71,357 @@ function print_help()
+
+            $mainIniFilePaths = find_main_ini_files($phpProps);
+
+            $set = false;
+            foreach ($mainIniFilePaths as $iniFile) {
+                if (!file_exists($iniFile)) {
+                    echo "File '$iniFile' does not exist. Trying to create it... ";
+                    if (file_put_contents($iniFile, '') === false) {
+                        echo "Could not set '{$setting[0]}' to '{$setting[1]}' in INI file: {$iniFile}.\n";
+                        $directory = dirname($iniFile);
+                        if (!is_dir($directory)) {
+                            echo "Directory '$directory' doesn't exist. Create it and try again if you want to use '$iniFile'.\n";
+                        }
+                        continue;
+                    } else {
+                        echo "Success.\n";
+                    }
+                }
+
+                $iniFileContent = file_get_contents($iniFile);
+                // check for "End of Line" symbol at the end of the file and
+                // add in case it is missing
+                if (strlen($iniFileContent) > 0 && substr($iniFileContent, -1, 1) !== PHP_EOL) {
+                    $iniFileContent .= PHP_EOL;
+                }
+                $iniFileContent .= implode(' = ', $setting) . PHP_EOL;
+                if (file_put_contents($iniFile, $iniFileContent) === false) {
+                    echo "Could not set '{$setting[0]}' to '{$setting[1]}' in INI file: {$iniFile}.\n";
+                    continue;
+                }
+                echo "Set '{$setting[0]}' to '{$setting[1]}' in INI file: $iniFile.\n";
+                $set = true;
+            }
+
+            if (!$set) {
+                echo "Unable to set '{$setting[0]}' to '{$setting[1]}' in any INI file for $binaryForLog.\n";
+                exit(1);
+            }
+        }
+    }
+}
+
+/**
+ * Parse a given INI setting (from CLI) into an array and return it. It also tries
+ * to parse the new setting with `parse_ini_string()` to validate it is actually
+ * working
+ *
+ * @return false|array{0:string, 1:string}
+ */
+function parse_ini_setting(string $setting)
+{
+    // `trim()` should not be needed, but better safe than sorry
+    $setting = array_map(
+        'trim',
+        explode(
+            '=',
+            $setting,
+            2
+        )
+    );
+    if (count($setting) !== 2) {
+        return false;
+    }
+    // safety: try out if parsing the generated ini setting is actually possible
+    if (parse_ini_string($setting[0] . '=' . $setting[1], false, INI_SCANNER_RAW) === false) {
+        return false;
+    }
+    return $setting;
+}
+
+
+/**
+ * This function will try and update a given `$setting` in the INI file given in
+ * `$iniFile`. First it tries to find an uncommented version of the setting in
+ * the file and replace this with the new value. In case no uncommented version
+ * was found it tries to find commented versions of this INI setting.
```

Comment:
That's not true, right?
If I read the function code correctly, it either replaces the first match, commented or not, or if $promoteComment is false, it replaces all occurrences of that INI.

Reply:
Also note that, if any directive happens to be duplicate, the last occurrence of that directive is relevant. It does not make sense to only replace the first, esp. in case there is a commented directive, followed by an uncommented one.

Reply:
Ah, I see, the caller site of update_ini_setting is ensuring it's first called without $promoteComment, then with it.
I guess in that case it's fine.

I was just looking at the function itself.

Reply:
Nevertheless, I will test this behaviour again to make sure there is no blind spot, thanks for the tip!